src/main/java/uk/gov/hmcts/reform/jobscheduler/jobs/HttpCallJob.java

```diff
@@ -32,6 +40,8 @@ public void execute(JobExecutionContext context) {
 
         HttpAction action = actionExtractor.extract(context);
 
+        action.headers.put("ServiceAuthorization", tokenGenerator.generate());
```

Comment:
Let's keep the action scheduled by the client what it was (instead of mutating it) and add this header when building a http request for RestTemplate.

Reply:
This is necessary because we can't persist the token for security reason. Even if it was encrypted, it could expire by the time the action is executed.